src/main/webapp/js/datepicker.js

```diff
@@ -70,8 +70,12 @@ function triggerDatepickerOnClick(datepickerDivs) {
 }
 
 /**
+ * Compares startDate and publishDate and returns the earlier date as minDate,
+ * if publishDate is undefined or null, returns startDate as minDate.
  * @assumption: startDate has a valid value
- * @returns {Date} publishDate if it is valid and smaller than startDate, else startDate
+ * @param {Date} startDate
+ * @param {Date} publishDate
+ * @returns {Date}
  */
 function getMaxDateForVisibleDate(startDate, publishDate) {
```

Comment:
This function is named getMaxDateForVisibleDate - I find it confusing since it is actually returning the minimum of 2 input dates; unless the max date that can be picked is the minimum of start and published date. Its also doing something general - returning the min of its input. Might be better to rename it so that it doesn't refer to start/published date, but is more general.